tools/lg_tools.py

```
@@ -273,16 +267,136 @@ def _extract_text_from_html(content: bytes) -> str:
+# ``KnowledgeStore`` instance. Doing it this way (rather than module-
+# level globals) keeps tests isolated — they pass a temp store and get
+# a fresh tool list bound to it. Production constructs one store in
+# ``server.py`` and reuses the bound tools for the lifetime of the
+# process.
+
+
+_MEMORY_RECALL_MAX_K = 20
+_MEMORY_LIST_MAX_LIMIT = 200
+
+
+def _build_memory_tools(knowledge_store):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_build_memory_tools`

(ANN202)